domain/src/main/java/org/oppia/domain/topic/TopicController.kt

```diff
@@ -66,6 +78,58 @@ class TopicController @Inject constructor() {
     )
   }
 
+  fun retrieveQuestionsForSkillIds(skillIdsList: List<String>): DataProvider<List<Question>> {
+    return dataProviders.createInMemoryDataProviderAsync(QUESTION_DATA_PROVIDER_ID) {
```

Comment:
I don't think you actually need to use the Async version. The data provider callback always gets called in a background thread, so it's fine to perform long-running operations even for the non-async version. The async version only exists if you want to do extra bits with AsyncResult, or specifically need to call a suspend function. Neither seem needed here (if any of the code throws an exception, it will automatically be turned into a failed AsyncResult downstream by DataProviders).

Reply:
I was loading the questions in a suspend function but I have removed that and kept this as the non-async version